opendevin/server/agent/manager.py

```diff
@@ -34,11 +32,7 @@ async def dispatch(self, sid: str, action: str | None, data: dict):
 
         await self.sid_to_agent[sid].dispatch(action, data)
 
-    def handle_signal(self, signum, _):
-        print(f'Received signal {signum}, exiting...')
-        self.close()
```

Comment:
This is not important anymore, but just FYI: this is a bug. Upon signal handler being called, `self.close()` is called here. Then at exit, `self.close` is called again by `atexit`.